javascript/unit-tests/KeySet.test.ts

```diff
@@ -0,0 +1,191 @@
+    const union = await ks.union(["key4", "key5", "key6"]);
+    ensure(union.has("key2"));
+    ensure(union.has("key5"));
+    ensure(!union.has("key9"));
+
+    // testing intersection
+    const intersection = await ks.intersection(["key3", "key4", "key5"]);
+    ensure(intersection.has("key3"));
+    ensure(!intersection.has("key2"));
+    ensure(!intersection.has("key4"));
+
+    // testing symmetric difference
+    const symDiff = await ks.symmetricDifference(["key3", "key4", "key5"]);
+    ensure(!symDiff.has("key3"));
+    ensure(symDiff.has("key2"));
+    ensure(symDiff.has("key4"));
+
+    // testing difference
+    const difference = await ks.difference(["key3", "key4", "key5"]);
+    ensure(difference.has("key1"));
+    ensure(difference.has("key2"));
+    ensure(!difference.has("key3"));
+});
+
+test('KeySet.toJson', async function() {
+    const instance = new GinkInstance(new IndexedDbStore('test6', true));
+    const ks = await instance.createKeySet();
+
+    await ks.add("key1");
+    await ks.update(["key2", "key3"]);
+
+    const asJSON = await ks.toJson();
+    ensure(asJSON == `["key1","key2","key3"]`, asJSON);
+});
+
+test('KeySet.asOf', async function() {
+    const instance = new GinkInstance(new IndexedDbStore('test7', true));
+    const ks = await instance.createKeySet();
+
+    const time0 = Date.now() * 1000;
+    await sleep(10);
+    await ks.add("key1");
+    await sleep(10);
+    const time1 = Date.now() * 1000;
+    await sleep(10);
+    await ks.add("key2");
+    await sleep(10);
+    const time2 = Date.now() * 1000;
+
+    // testing asOf for has
+    ensure(await ks.has("key1", time1));
+    ensure(await ks.has("key1", -1));
+    ensure(await ks.has("key1", time2));
+    ensure(!await ks.has("key2", time1));
+
+    // testing asOf for toJson
+    ensure(await ks.toJson(false, time1)==`["key1"]`);
+    ensure(await ks.toJson(false, time2)==`["key1","key2"]`);
+
+    // testing asOf for size
+    ensure(await ks.size(time0)==0);
+    ensure(await ks.size(time1)==1);
+    ensure(await ks.size(time2)==2);
+
+    // testing asOf toSet
+    const values = await ks.toSet(time0);
+    const values1 = await ks.toSet(time1);
+
+    ensure(!values.size);
+    ensure(!values.has("key2"));
+    ensure(values1.size == 1);
+    ensure(values1.has("key1"));
+});
+
+test('KeySet.clear', async function() {
+    const instance = new GinkInstance(new IndexedDbStore('test8', true));
+    const ks = await instance.createKeySet();
+    await ks.update(["key1", "key2"]);
+    const clearMuid = await ks.clear();
+    await ks.update(["key3", "key4"]);
+    const asSet = await ks.toSet();
+    ensure(asSet.has("key4") && !asSet.has("key1"), "did not clear")
+    const asSetBeforeClear = await ks.toSet(clearMuid.timestamp);
+    if (asSetBeforeClear.has("key4") || !asSetBeforeClear.has("key1")) {
+        console.log(asSetBeforeClear);
+        throw new Error("busted");
+    }
+});
+
+test('KeySet.purge', async function () {
```

Comment:
You're not actually testing the purge capability in this test. (Purge being a flag that you would pass to a delete or clear that would force removal of history so it would prevent things from showing up in as-of queries)